frontend/src/components/homepage/homepage.js

```diff
@@ -1,14 +1,16 @@
-import React, { useState } from 'react';
-import TravelForm from '../travelForm/TravelForm';
-import EmissionResults from '../emissionResults/emissionResults';
-import NavBar from '../navBar/navBar';
-
-const HomePage = ({navigate}) => {
+import React, { useState } from "react";
+import TravelForm from "../travelForm/TravelForm";
+import EmissionResults from "../emissionResults/emissionResults";
+import NavBar from "../navBar/navBar";
 
+const HomePage = ({ navigate }) => {
   const [emissions, setEmissions] = useState("");
   const [renderEmissions, setRenderEmissions] = useState(false);
   const [toDisplay, setToDisplay] = useState("");
   const [fromDisplay, setFromDisplay] = useState("");
+  const [passengers, setPassengers] = useState("");
+  const [saveToggle, setSaveToggle] = useState(false);
+  const [saveText, setSaveText] = useState("SAVE");
```

Comment:
You could just have an if statement that changes the text of the button based on `saveToggle`